From 34f93ea4294e2254c95478521fa9886dfed43910 Mon Sep 17 00:00:00 2001
From: Ivan Solovev <ivan.solovev@qt.io>
Date: Mon, 21 Aug 2023 13:15:16 +0200
Subject: QtPdf: migrate to autogenerated cpp exports

This is a prerequisite for using QT_PDF_REMOVED_SINCE.

Pick-to: 6.6
Change-Id: I46adece3ba1ff1a4c95218a1bc731dea05afc1f3
Reviewed-by: Alexey Edelev <alexey.edelev@qt.io>
Reviewed-by: Shawn Rutledge <shawn.rutledge@qt.io>
Reviewed-by: Marc Mutz <marc.mutz@qt.io>
---
 src/pdf/CMakeLists.txt   |  2 ++
 src/pdf/qpdfdocument_p.h |  1 +
 src/pdf/qtpdfglobal.h    | 19 +------------------
 3 files changed, 4 insertions(+), 18 deletions(-)

diff --git a/src/pdf/CMakeLists.txt b/src/pdf/CMakeLists.txt
index f51e461ac..8070eb39a 100644
--- a/src/pdf/CMakeLists.txt
+++ b/src/pdf/CMakeLists.txt
@@ -38,6 +38,8 @@ qt_internal_add_module(Pdf
     PUBLIC_LIBRARIES
         Qt::Core
         Qt::Gui
+    GENERATE_CPP_EXPORTS
+    GENERATE_PRIVATE_CPP_EXPORTS
 )
 
 add_subdirectory(plugins/imageformats/pdf)
diff --git a/src/pdf/qpdfdocument_p.h b/src/pdf/qpdfdocument_p.h
index d52aabe4b..4adbf695e 100644
--- a/src/pdf/qpdfdocument_p.h
+++ b/src/pdf/qpdfdocument_p.h
@@ -16,6 +16,7 @@
 //
 
 #include "qpdfdocument.h"
+#include "private/qtpdfexports_p.h"
 
 #include "third_party/pdfium/public/fpdfview.h"
 #include "third_party/pdfium/public/fpdf_dataavail.h"
diff --git a/src/pdf/qtpdfglobal.h b/src/pdf/qtpdfglobal.h
index d38eafaab..2d0900029 100644
--- a/src/pdf/qtpdfglobal.h
+++ b/src/pdf/qtpdfglobal.h
@@ -5,24 +5,7 @@
 #define QTPDFGLOBAL_H
 
 #include <QtCore/qglobal.h>
-
-QT_BEGIN_NAMESPACE
-
-#ifndef Q_PDF_EXPORT
-#  ifndef QT_STATIC
-#    if defined(QT_BUILD_PDF_LIB)
-#      define Q_PDF_EXPORT Q_DECL_EXPORT
-#    else
-#      define Q_PDF_EXPORT Q_DECL_IMPORT
-#    endif
-#  else
-#    define Q_PDF_EXPORT
-#  endif
-#endif
-
-#define Q_PDF_PRIVATE_EXPORT Q_PDF_EXPORT
-
-QT_END_NAMESPACE
+#include <QtPdf/qtpdfexports.h>
 
 #endif // QTPDFGLOBAL_H
 
-- 
cgit v1.2.3

